### ISSUE TITLE
Chain behaviours for the same command/payload

### ISSUE THREAD

**Issue by angarg12** [m-radzikowski/aws-sdk-client-mock#17]
It isn't unusual in Jest to chain behaviours of a mock, for instance with several sequential calls to `mockImplementationOnce`. This useful when you want to reproduce some specific behaviour, such as a call succeeding the first time but failing the second time.

I ran into this problem when I was refactoring my code to use this library. I was able to overcome it by refactoring my code a bit and giving each call/responses unique Ids. 

I would consider this lower priority since in most cases you should be able to work around it. Still it might help to simplify test code.

**Comment by m-radzikowski** [comment]
Yes, that should be doable. Right now you could do this manually by instrumenting the Sinon.js stub that is used under the hood. But adding chaining to the library seems like a nice feature.

**Comment by nguyen102** [comment]
Is there an example of how this can be done with Sinon stubs?

**Comment by dmattia** [comment]
Sure, I do this like:

```typescript
    const dbMock = mockClient(someClient);

    // The first call should return LastEvaluatedKey, signaling that pagination should happen
    dbMock.send.onFirstCall().resolves({
      Items: [{ pk: { S: 'a' }}],
      LastEvaluatedKey: { pk: { S: 'a'} },
    });

    // The second call is the final call
    dbMock.send.onSecondCall().resolves({
      Items: [{ pk: { S: 'b' }} ] ,
    });
```

**Comment by nguyen102** [comment]
That worked perfectly. Thanks so much @dmattia

**Comment by andres-ntropy** [comment]
I've been using this approach but lately it stopped working (with TypeScript)

```
const sqs = mockClient(SQSClient);

    sqs.send
      .onFirstCall()
      .resolves({ Successful: Array(10).fill({ Id: "a" }) })
      .onSecondCall()
      .resolves({
        Failed: [
          { Id: "f1", Code: "FAIL", SenderFault: true },
          { Id: "f2", Code: "FAIL", SenderFault: true },
        ],
      })
      .onThirdCall()
      .resolves({ Successful: [{ Id: "f1" }, { Id: "f2" }] });
```

It fails with: _Argument of type '{ Successful: any[]; }' is not assignable to parameter of type 'ServiceOutputTypes'._

**Comment by delsinz** [comment]
> Sure, I do this like:
> 
> ```ts
>     const dbMock = mockClient(someClient);
> 
>     // The first call should return LastEvaluatedKey, signaling that pagination should happen
>     dbMock.send.onFirstCall().resolves({
>       Items: [{ pk: { S: 'a' }}],
>       LastEvaluatedKey: { pk: { S: 'a'} },
>     });
> 
>     // The second call is the final call
>     dbMock.send.onSecondCall().resolves({
>       Items: [{ pk: { S: 'b' }} ] ,
>     });
> ```

This works perfectly, but what if I want to chain more than three calls on the same client? Looks like `onThirdCall` is as far as it gets. Not sure if there's a workaround.

**Comment by m-radzikowski** [comment]
It took a moment but is now released in v0.6.2 🎉 

See the new `resolvesOnce()`, `rejectsOnce()`, and `callsFakeOnce()` mock functions.
